Library/Homebrew/utils/popen.rb

```diff
@@ -12,8 +12,27 @@ def self.safe_popen_read(*args, **options, &block)
     raise ErrorDuringExecution.new(args, status: $CHILD_STATUS, output: [[:stdout, output]])
   end
 
-  def self.popen_write(*args, **options, &block)
-    popen(args, "wb", options, &block)
+  def self.popen_write(*args, **options)
+    popen(args, "w+b", options) do |pipe|
+      output = +""
+      buffer = +""
```

Comment:
Does this definitely need to be a non-mutable string?

Reply:
(I assume you meant to ask *Does this definitely need to be a mutable string?*)

Using a buffer is not really important here so I’m fine with dropping the buffer.

Reply:
Yeh, makes sense to me 👍🏻